targets/TARGET_Analog_Devices/TARGET_ADUCM302X/TARGET_ADUCM3029/api/trng_api.c

```diff
@@ -55,7 +55,7 @@
 #define TRNG_PRESCALER  2
 
 /* RNG Device memory */
-static uint8_t RngDevMem[ADI_RNG_MEMORY_SIZE];
+static uint32_t RngDevMem[(ADI_RNG_MEMORY_SIZE + 3)/4];
```

Comment:
This does not seem to be related to armc6, shall be separate commit

Reply:
The default alignment is set to 1 byte by ARMC6. uint32_t gurantee 4 byte alignment.
static uint8_t RngDevMem[ADI_RNG_MEMORY_SIZE]; exactly causes ARMC6 greentea tests fail in
test case mbed-os-tests-mbedtls-selftest.
static uint32_t RngDevMem[(ADI_RNG_MEMORY_SIZE+3)/4]; pass in test case mbed-os-tests-mbedtls-selftest for both ARMC6 and ARMC5.

The code is meant be compatible for both ARMC6 and ARMC5.

Reply:
Thanks for the explanation, please use commit msg to provide more detailed description for changes. If you can, I would ammend the commit and add there details for changes.

Reply:
2 commits added